src/MeshInertiaCalculator.cc

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <optional>
+#include <vector>
+
+#include <sdf/sdf.hh>
+#include <sdf/CustomInertiaCalcProperties.hh>
+
+#include <gz/sim/Util.hh>
+
+#include <gz/common/graphics.hh>
+#include <gz/common/Mesh.hh>
+#include <gz/common/MeshManager.hh>
+
+#include <gz/math/Vector3.hh>
+#include <gz/math/MassMatrix3.hh>
+#include <gz/math/Inertial.hh>
+
+#include <gz/sim/MeshInertiaCalculator.hh>
+
+using namespace gz;
+using namespace sim;
+
+//////////////////////////////////////////////////
+MeshInertiaCalculator::MeshInertiaCalculator() = default;
+
+//////////////////////////////////////////////////
+MeshInertiaCalculator::~MeshInertiaCalculator() = default;
+
+//////////////////////////////////////////////////
+void MeshInertiaCalculator::GetMeshTriangles(
+    std::vector<Triangle>& _triangles,
+    const gz::common::Mesh* _mesh)
+{
+  // Get the vertices & indices of the mesh
+  double* vertArray = NULL;
+  int* indArray = NULL;
```

Comment:
use `nullptr` instead of `NULL`

Reply:
Replaced in `NULL` with `nullptr` in [a2a1cba](https://github.com/gazebosim/gz-sim/pull/2061/commits/a2a1cba22551449b3afa2c276d0e6bd02271f780)